### ISSUE TITLE
Issue submitted via apport

### ISSUE THREAD

**Issue by ruenoak** [ubuntu-flutter-community/musicpod#370]
**ProblemType**
Bug

**Date**
Wed Dec 20 19:59:26 2023

**CurrentDesktop**
XFCE

**ProcEnviron**
LANG=en_NZ.UTF-8
LANGUAGE=en_NZ:en
PATH=(custom, no user)
SHELL=/bin/bash
XDG_RUNTIME_DIR=<set>

**DistroRelease**
Ubuntu 24.04

**Uname**
Linux 6.5.0-9-generic x86_64

**Architecture**
amd64

**Snap**
musicpod 0+git.3376452 (latest/stable)

**SnapChanges**
ID   Status  Spawn                      Ready                      Summary
16   Done    2023-12-20T19:43:37+13:00  2023-12-20T19:44:32+13:00  Refresh snaps "core20", "firefox", "musicpod"

**SnapConnections**
Interface                Plug                              Slot                            Notes
audio-playback           musicpod:audio-playback           :audio-playback                 -
content[gnome-42-2204]   musicpod:gnome-42-2204            gnome-42-2204:gnome-42-2204     -
content[gtk-3-themes]    musicpod:gtk-3-themes             gtk-common-themes:gtk-3-themes  -
content[icon-themes]     musicpod:icon-themes              gtk-common-themes:icon-themes   -
content[sound-themes]    musicpod:sound-themes             gtk-common-themes:sound-themes  -
dbus                     -                                 musicpod:dbus-musicpod          -
desktop                  musicpod:desktop                  :desktop                        -
desktop-legacy           musicpod:desktop-legacy           :desktop-legacy                 -
gsettings                musicpod:gsettings                :gsettings                      -
home                     musicpod:home                     :home                           -
mpris                    -                                 musicpod:mpris                  -
network                  musicpod:network                  :network                        -
network-manager-observe  musicpod:network-manager-observe  :network-manager-observe        -
opengl                   musicpod:opengl                   :opengl                         -
removable-media          musicpod:removable-media          -                               -
wayland                  musicpod:wayland                  :wayland                        -
x11                      musicpod:x11                      :x11                            -

**SnapInfo.musicpod**
name:      musicpod
summary:   Music, podcast and internet radio player for your Ubuntu Desktop
publisher: Frederik Feichtmeier
store-url: https://snapcraft.io/musicpod
contact:   https://github.com/ubuntu-flutter-community/musicpod
license:   MPL-2.0
description: |
  Play local audio files, browse podcasts online or listen to internet radio
  stations - all in one app!
  
  - Play local audio files (mp3, flac, m4a, mp4)
  - Find your local audio library sorted by metadata
  - Search for local audio files sorted by title, album and artist
  - Play internet radio stations and search them by tag, name or country
  - Play podcasts and find them by country, topic and name
  - Pin your favorite albums, podcasts and stations to the sidebar
  - Collect your favorite songs with the heart button
  
  The podcast and radio features do currently need internet access and thus
  the network-manager-observe permission is being used to display a
  disconnect from the internet in the user interface.
  
  MusicPod should automatically connect to the network-manager-observe
  interface. If it didn't you can connect it with the following command:
  
  `sudo snap connect musicpod:network-manager-observe`
commands:
  - musicpod
snap-id:      sxdOSGRbzsWhPmGrkYYAs68rzieTrIiv
tracking:     latest/stable
refresh-date: 2023-12-20T19:43:53+13:00
channels:
  latest/stable:    0+git.3376452 2023-12-19T16:34:17Z (689) 54MB -
  latest/candidate: ^                                             
  latest/beta:      ^                                             
  latest/edge:      0+git.d68b6de 2023-12-20T06:48:03Z (693) 54MB -
installed:          0+git.3376452                      (689) 54MB -

**SnapInfo.core22**
name:      core22
summary:   Runtime environment based on Ubuntu 22.04
publisher: Canonical**
store-url: https://snapcraft.io/core22
contact:   https://github.com/snapcore/core-base/issues
license:   unset
description: |
  The base snap based on the Ubuntu 22.04 release.
type:         base
snap-id:      amcUKQILKXHHTlmSa7NMdnXSx02dNeeT
tracking:     latest/stable
refresh-date: 2023-12-12T19:32:01+13:00
channels:
  latest/stable:          20231123 2023-12-09T03:10:03Z (1033) 77MB -
  latest/candidate:       20231123 2023-12-05T16:41:55Z (1033) 77MB -
  latest/beta:            20231123 2023-11-27T16:35:20Z (1033) 77MB -
  latest/edge:            20231211 2023-12-12T00:05:53Z (1075) 77MB -
  fips-updates/stable:    20231019 2023-11-03T07:27:10Z  (952) 78MB -
  fips-updates/candidate: ^                                         
  fips-updates/beta:      ^                                         
  fips-updates/edge:      ^                                         
  fips-preview/stable:    20231019 2023-11-03T07:24:05Z  (952) 78MB -
  fips-preview/candidate: ^                                         
  fips-preview/beta:      ^                                         
  fips-preview/edge:      ^                                         
  fips/stable:            --                                        
  fips/candidate:         --                                        
  fips/beta:              --                                        
  fips/edge:              20231019 2023-11-03T07:27:35Z  (952) 78MB -
installed:                20231123                      (1033) 77MB base

**SnapInfo.gtk-common-themes**
name:      gtk-common-themes
summary:   All the (common) themes
publisher: Canonical**
store-url: https://snapcraft.io/gtk-common-themes
license:   unset
description: |
  A snap that exports the GTK and icon themes used on various Linux distros.
snap-id:      jZLfBRzf1cYlYysIjD2bwSzNtngY0qit
tracking:     latest/stable/ubuntu-24.04
refresh-date: 2023-11-18T19:24:24+13:00
channels:
  latest/stable:    0.1-81-g442e511 2022-06-28T18:10:06Z (1535) 96MB -
  latest/candidate: 0.1-81-g442e511 2022-06-24T16:30:46Z (1535) 96MB -
  latest/beta:      0.1-81-g442e511 2022-06-24T16:30:46Z (1535) 96MB -
  latest/edge:      ^                                                
installed:          0.1-81-g442e511                      (1535) 96MB -

**SnapInfo.gnome-42-2204**
name:      gnome-42-2204
summary:   Shared GNOME 42 Ubuntu stack
publisher: Canonical**
store-url: https://snapcraft.io/gnome-42-2204
contact:   https://github.com/ubuntu/gnome-sdk/issues
license:   unset
description: |
  This snap provides the GNOME 42 stack to other snaps that use it. It shares
  the base GNOME libraries and desktop integration components through the
  content interface. This helps reduce the size of snaps and helps developers
  to easily snap desktop applications.
  
  **For users**
  
  This snap is automatically installed and removed when needed. **Manually
  adding or removing this snap is not recommended** and might break things.
  
  * If you are having issues with **snaps** using GNOME, please contact the
  experts on the Snapcraft forum: https://forum.snapcraft.io/
  * If you want to install the GNOME Desktop Environment, then you are in the
  wrong place. Please take a look at https://www.gnome.org/ for more
  information on how to get it.
  
  **For developers**
  
  * The `gnome` extension is the recommended way to use this in your own
  snap: https://snapcraft.io/docs/gnome-extension
  * You can report issues with this content snap on GitHub:
  https://github.com/ubuntu/gnome-sdk/issues
  * The source code of this snap is available on GitHub in the
  `gnome-42-2204` branch:
  https://github.com/ubuntu/gnome-sdk/tree/gnome-42-2204
snap-id:      lATO8HzwVvrAPrlZRAWpfyrJKlAJrZS3
tracking:     latest/stable/ubuntu-24.04
refresh-date: 2023-11-18T19:24:24+13:00
channels:
  latest/stable:    0+git.ff35a85 2023-09-30T05:10:04Z (141) 521MB -
  latest/candidate: 0+git.bf328d8 2023-11-11T02:51:00Z (158) 524MB -
  latest/beta:      ^                                              
  latest/edge:      0+git.bf328d8 2023-11-11T02:51:00Z (158) 524MB -
installed:          0+git.ff35a85                      (141) 521MB -

**SnapGitOwner**
ubuntu-flutter-community

**SnapGitName**
musicpod

**CrashDB**
snap-github

**InstallationMedia**
Xubuntu 24.04 "Noble Numbat" - Daily amd64 (20231109)

**Tags**
noble

**InstallationDate**
Installed on 2023-11-18 (32 days ago)

**UpgradeStatus**
No upgrade log present (probably fresh install)

**ProcVersionSignature**
Ubuntu 6.5.0-9.9-generic 6.5.3

**ProcCpuinfoMinimal**
processor	: 3
vendor_id	: AuthenticAMD
cpu family	: 18
model		: 1
model name	: AMD A6-3420M APU with Radeon(tm) HD Graphics
stepping	: 0
microcode	: 0x3000027
cpu MHz		: 799.582
cache size	: 1024 KB
physical id	: 0
siblings	: 4
core id		: 3
cpu cores	: 4
apicid		: 3
initial apicid	: 3
fpu		: yes
fpu_exception	: yes
cpuid level	: 6
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm 3dnowext 3dnow constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf pni monitor cx16 popcnt lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt cpb hw_pstate vmmcall arat npt lbrv svm_lock nrip_save pausefilter
bugs		: fxsave_leak sysret_ss_attrs null_seg spectre_v1 spectre_v2
bogomips	: 2994.44
TLB size	: 1536 4K pages
clflush size	: 64
cache_alignment	: 64
address sizes	: 40 bits physical, 48 bits virtual
power management: ts ttp tm stc 100mhzsteps hwpstate cpb

**ApportVersion**
2.27.0-0ubuntu6

**CasperMD5CheckResult**
pass

**Comment by ruenoak** [comment]
When opening Musicpod I get a blank window.
opening via the terminal I get the following:
(musicpod:4155): Gdk-CRITICAL **: 19:53:56.686: gdk_window_get_state: assertion 'GDK_IS_WINDOW (window)' failed
package:media_kit_libs_linux registered.

(musicpod:4155): IBUS-WARNING **: 19:53:56.753: Failed to mkdir /home/rue/snap/musicpod/689/.config/ibus/bus: Not a directory
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: r600
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: r600
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast

** (musicpod:4155): WARNING **: 19:53:57.257: Failed to start Flutter renderer: Unable to create a GL context

** (musicpod:4155): WARNING **: 19:54:22.269: atk-bridge: get_device_events_reply: unknown signature

**Comment by Feichtmeier** [comment]
Thanks! 
This could be reasoned by your relatively new kernel and flutter not properly rendering against this.

Could you try to open the snap-store or the firmware-updater snap and see if this works?

**Comment by ruenoak** [comment]
Hi I installed the Snap-store and the firmware-updater, snaps. 
The firmware-updater has the same issue with just a blank window opening, When I try the snap-store it opens as normal and seem to be working.

**Comment by Feichtmeier** [comment]
> Hi I installed the Snap-store and the firmware-updater, snaps. The firmware-updater has the same issue with just a blank window opening, When I try the snap-store it opens as normal and seem to be working.

thanks and I assume the snap-store snap is not the old gnome based version but the new flutter based version?
just post a screenshot of the app center (snap-store) for me to check

**Comment by ruenoak** [comment]
![Screenshot_2023-12-21_18-37-31](https://github.com/ubuntu-flutter-community/musicpod/assets/101319803/7316eade-a940-4cee-8b46-16c113da1dbf)
Hi, I installed it from the snap store, below is from 'snap list' It seems to be the latest stable from the store.
snap-store         41.3-71-g709398e   959    latest/stable    canonical✓            -

**Comment by Feichtmeier** [comment]
> ![Screenshot_2023-12-21_18-37-31](https://github.com/ubuntu-flutter-community/musicpod/assets/101319803/7316eade-a940-4cee-8b46-16c113da1dbf)
> Hi, I installed it from the snap store, below is from 'snap list' It seems to be the latest stable from the store.
> snap-store         41.3-71-g709398e   959    latest/stable    canonical✓            -
> 

Thanks! Yeah that's the gnome based store. 
I don't want you to change your system in a way you don't like but the flutter made store is currently in the channel latest/edge
You could change to this with

```
killall snap-store 

sudo snap refresh snap-store --channel=latest/edge
```

And to switch back

```
sudo snap refresh snap-store --channel=latest/stable
```

**Comment by ruenoak** [comment]
Not a problem, I don't mind changing the system, this is my Xubuntu noble testing machine so it is all fine, I will install the edge version and report back.

**Comment by ruenoak** [comment]
I installed the edge version of the snap-store and confirm that I am getting the same problem as the the Firmware-upater and Musitpod. So I think as you said it's a flutter new kernel issue. 
I could submit this as a bug against flutter on Launchpad but I think this is not a bug but more that updates just haven't synced up yet in noble.

**Comment by Feichtmeier** [comment]
hey @ruenoak 
is this fixed in 24.04 now?

**Comment by ruenoak** [comment]
HI yes it is, Music pod, App centre  work as expected now. 

Thanks
Rue.
